src/preamble.js

```diff
@@ -2220,6 +2220,9 @@ function integrateWasmJS(Module) {
     } else if (Module['readBinary']) {
       binary = Module['readBinary'](wasmBinaryFile);
     } else {
+      binary = tryParseAsDataURI(wasmBinaryFile);
```

Comment:
It should work as-is since `Module.readBinary` handles parsing data URIs on any platform, but how about this?

```
function getBinary() {
  if (Module['wasmBinary']) {
    return new Uint8Array(Module['wasmBinary']);
  }
  var binary = tryParseAsDataURI(wasmBinaryFile);
  if (binary) {
    return binary;
  } else if (Module['readBinary']) {
    return Module['readBinary'](wasmBinaryFile);
  } else {
    throw "on the web, we need the wasm binary to be preloaded and set on Module['wasmBinary']. emcc.py will do that for you when generating HTML (but not JS)";
  }
}
```

(I'm assuming `Module.wasmBinary` should take priority over `tryParseAsDataURI(wasmBinaryFile)`, even though it doesn't really matter in practice right now.)